submissions/api.py

```diff
@@ -684,6 +685,12 @@ def reset_score(student_id, course_id, item_id):
             item_id=item_id,
         )
 
+        if clear_state:
+            # sever the link between this student item and any submissions it may current have
+            for sub in student_item.submission_set.all():
+                sub.student_item = None
+                sub.save(update_fields=['student_item'])
```

Comment:
I'm not sure how thorough the event coverage is, but see my latest comment on https://github.com/edx/edx-submissions/pull/35. We may be able to keep the original student association mostly intact.